Libraries/LibChess/Chess.cpp

```diff
@@ -24,13 +24,60 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "Chess.h"
 #include <AK/Assertions.h>
 #include <AK/LogStream.h>
+#include <AK/String.h>
+#include <AK/StringBuilder.h>
 #include <AK/Vector.h>
+#include <LibChess/Chess.h>
 #include <stdlib.h>
 
-Chess::Square::Square(const StringView& name)
+namespace Chess {
+
+String char_for_piece(Chess::Type type)
+{
+    switch (type) {
+    case Type::Knight:
+        return "N";
+    case Type::Bishop:
+        return "B";
+    case Type::Rook:
+        return "R";
+    case Type::Queen:
+        return "Q";
+    case Type::King:
+        return "K";
+    case Type::Pawn:
+    default:
+        return "";
+    }
+}
+
+Chess::Type piece_for_char(const StringView& str)
+{
+    String string = String(str).to_lowercase();
+    if (string == "")
+        return Type::None;
```

Comment:
Possibly a typo ? :P
```suggestion
    if (string == "")
        return Type::Pawn;
```

Reply:
This is not a typo, but I see how it could be confusing. `piece_for_char` is used only for promotions, where no provided piece give `promote_to = Type::None`

Reply:
Oh, that makes sense ! And yeah, definitely, especially since its behaviour is not symmetric with `char_for_piece` (having `piece_for_char(char_for_piece(Type::Pawn)) == Type::Null` is very counter-intuitive)